### PR TITLE
DisplayStyle API cleanup/promotion

### PR DESCRIPTION
**Pull request by pmconne**
A DisplayStyleSettings has subcategory appearance overrides, planar clip mask settings, and model appearance overrides, all implemented similarly with somewhat tricky copy-pasted code to keep in sync with JSON representation.
Implement all in terms of a Map that automatically synchronizes the JSON representation.
Document and promote all to public.
Also clean up and promote APIs for planar clip masks, context reality models, and spatial classification.
See changes to NextVersion.md for a summary.

~~TODO: No tests exist for any of the existing APIs. Add some.~~ Added.
~~TODO: The Map API suffices (set, delete, get, has) but half a dozen helper methods also exist; deprecate them?~~ Not now.
~~TODO: ContextRealityModel-related APIs need work.~~ Done.

### CODE REVIEW

File: core/common/src/PlanarClipMask.ts [hunk 1]
```diff
@@ -47,71 +42,85 @@ export enum PlanarClipMaskPriority {
...
-  /** The SubCategory ids if mode is [[PlanarClipMaskMode.IncludeSubCategorie]] or element Ids if mode is [[PlanarClipMaskMode.IncludeElements] or [[PlanarClipMaskMode.ExcludeElements]] */
+  /** For [[PlanarClipMaskMode.IncludeElements]] or [[PlanarClipMaskMode.ExcludedElements]], the Ids of the [GeometricElement]($backend)s to include or exclude from masking;
+   * for [[PlanarClipMaskMode.IncludeSubCategories]], the Ids of the subcategories whose geometry contributes to the mask.
+   */
   public readonly subCategoryOrElementIds?: CompressedId64Set;
-  /** Priority value if mode is [[PlanarClipMaskMode.Priority]]. */
+  /** For [[PlanarClipMaskMode.Priority]], the priority value. */
   public readonly priority?: number;
-  /** A value between 0 and 1 indicating am override for mask transparency - A transparency of 0 (the default) indicates complete masking.  1 is completely transparent (no masking).  If no transparency is defined then the transparency of the mask elements are used */
+  /** A value between 0 and 1 indicating an override for mask transparency. A transparency of 0 (the default) indicates complete masking. 1 is completely transparent (no masking).
```
Comment (pmconne) [inline core/common/src/PlanarClipMask.ts:90]: @RBBentley it says zero is the default but also that `undefined` means "use element transparency". Only the latter is true, correct?

Comment (RBBentley) [reply core/common/src/PlanarClipMask.ts:90]: Yes -- "(the default)" should be removed.

File: core/common/src/PlanarClipMask.ts [hunk 2]
```diff
@@ -47,71 +42,85 @@ export enum PlanarClipMaskPriority {
   /** A reality model with a bounded range. */
   RealityModel = 0,
   /** A design model stored in the [IModelDb]($backend). */
-  BIM = 2048,
+  DesignModel = 2048,
   /** The maximum supported priority. */
   Maximum = 4096,
```
Comment (pmconne) [inline core/common/src/PlanarClipMask.ts:47]: @RBBentley is this a hard limit for some technical reason, such that PlanarClipMaskSettings should be enforcing it?
Is there correspondingly a minimum priority?

Comment (RBBentley) [reply core/common/src/PlanarClipMask.ts:47]: There really isn't a limit -- probably could remove this.     The relative magnitude only is used.   In general these only time you would want to use anything but defaults is if you wanted to order the relative masking order of a set of reality models.

File: core/common/src/PlanarClipMask.ts [hunk 3]
```diff
@@ -47,71 +41,90 @@ export enum PlanarClipMaskPriority {
   /** A reality model with a bounded range. */
   RealityModel = 0,
   /** A design model stored in the [IModelDb]($backend). */
-  BIM = 2048,
-  /** The maximum supported priority. */
-  Maximum = 4096,
```
Comment (mgooding) [inline core/common/src/PlanarClipMask.ts:52]: Maximum intentionally removed?

File: core/common/src/PlanarClipMask.ts [hunk 4]
```diff
@@ -47,71 +41,90 @@ export enum PlanarClipMaskPriority {
...
+  DesignModel = 2048,
 }
 
 /** JSON representation of a [[PlanarClipMaskSettings]].
- * @beta
+ * @see [[DisplayStyleSettingsProps.planarClipOvr and [[ContextRealityModelProps.planarClipMask]].
```
Comment (mgooding) [inline core/common/src/PlanarClipMask.ts:48]: `[[DisplayStyleSettingsProps.planarClipOvr]]` instead?

File: core/frontend/src/Viewport.ts [hunk 5]
```diff
@@ -158,18 +158,6 @@ export interface ChangeViewedModel2dOptions {
  */
 export enum ViewUndoEvent { Undo = 0, Redo = 1 }
 
-/** Options controlling display of [OpenStreetMap Buildings](https://cesium.com/platform/cesium-ion/content/cesium-osm-buildings/).
- * @see [[Viewport.setOSMBuildingDisplay=]].
-   * @public
```
Comment (mgooding) [inline core/frontend/src/Viewport.ts:163]: Is this a breaking change for public API?

**Review by mgooding — APPROVED** [review]

File: core/frontend/src/Viewport.ts (hunk 5 above)
Comment (pmconne) [reply core/frontend/src/Viewport.ts:163]: It moved to DisplayStyleState.ts - is unused in Viewport.ts.

File: core/common/src/PlanarClipMask.ts (hunk 3 above)
Comment (pmconne) [reply core/common/src/PlanarClipMask.ts:52]: Yes, as discussed with Ray - nothing enforces/checks it.

File: core/common/src/PlanarClipMask.ts (hunk 4 above)
Comment (pmconne) [reply core/common/src/PlanarClipMask.ts]: Yup. Fixed.

File: core/frontend-devtools/src/tools/RealityModelTools.ts [hunk 6]
```diff
@@ -81,8 +81,12 @@ export class SaveRealityModelTool extends Tool {
 }
 
 function changeRealityModelAppearanceOverrides(vp: Viewport, overrides: FeatureAppearanceProps, index: number): boolean {
-  const existingOverrides = vp.getRealityModelAppearanceOverride(index);
-  return vp.overrideRealityModelAppearance(index, existingOverrides ? existingOverrides.clone(overrides) : FeatureAppearance.fromJSON(overrides));
+  const model = vp.displayStyle.settings.contextRealityModels.models[index];
+  if (!model)
```
Comment (RBBentley) [inline core/frontend-devtools/src/tools/RealityModelTools.ts:85]: The viewport listener now handles the invalidating render plan etc. so the viewport methods are superfluous, right?

Comment (pmconne) [reply core/frontend-devtools/src/tools/RealityModelTools.ts:85]: Correct. Viewport had ones that called DisplayStyleState ones that called DisplayStyleSettings ones...I removed the redundant non-`public` ones.

File: core/frontend/src/Viewport.ts (hunk 5 above)
Comment (pmconne) [reply core/frontend/src/Viewport.ts:163]: Please resolve comments if addressed to your satisfaction - PRs cannot be completed until all comments are resolved.

**Review by kabentley — APPROVED** [review]

**Review by bbastings — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

**Review by ColinKerr — APPROVED** [review]

**Pull request merged**